_posts/2021-09-13-user-defined-value-types.md

```diff
@@ -0,0 +1,152 @@
+---
+layout: post
+published: true
+title:  User Defined Value Types in Solidity
+date:   2021-09-13
+author: Solidity Team
+category: Explainers
+---
+
+Solidity [Solidity v0.8.8](https://github.com/ethereum/solidity/releases/tag/v0.8.8) introduces
+[user defined value
+types](https://docs.soliditylang.org/en/develop/types.html?#user-defined-value-types) as a means to
+create zero-cost abstractions over an elementary value type.
+
+## Motivation
+
+A problem with primitive value types is that they are not very descriptive. For example, one may
+want to use `uint128` to store the price of some object as well as the quantity available. Having
+stricter type rules to avoid intermingling of the two different concepts can also be quite useful.
+For example, one may want to disallow assigning a quantity to a price or vice versa.
+
+One option for solving this issue is by using
+[structs](https://docs.soliditylang.org/en/develop/types.html#structs). For example, price and
+quantity can be abstracted as structs as follows:
+
+```solidity
+struct Price { uint128 price; }
+struct Quantity { uint128 quantity; }
+
+function toPrice(uint128 price) returns(Price memory) {
+    return Price(price);
+}
+function fromPrice(Price memory price) returns(uint128) {
+   return price.price;
+}
+function toQuantity(uint128 quantity) returns(Quantity memory) {
+    return Quantity(quantity);
+}
+function fromQuantity(Quantity memory quantity) returns(uint128) {
+    return quantity.quantity;
+}
+```
+
+However, a [struct](https://docs.soliditylang.org/en/latest/types.html?#structs) is a reference type
+and therefore always points to a value in `memory`, `calldata` or `storage`. This means that the
```

Comment:
I remember that somehow it is always better to use double backticks in rst, but I don't remember why.

Reply:
But this is just plain markdown file.